SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDevAuthTokenGenerator.cpp

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKDevAuthTokenGenerator.h"
+
+#include "Async/Async.h"
+#include "Framework/Notifications/NotificationManager.h"
+
+#include "SpatialCommandUtils.h"
+#include "SpatialGDKEditorSettings.h"
+#include "SpatialGDKSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKDevAuthTokenGenerator);
+
+FSpatialGDKDevAuthTokenGenerator::FSpatialGDKDevAuthTokenGenerator()
+	: bIsGenerating(false)
+{
+}
+
+void FSpatialGDKDevAuthTokenGenerator::UpdateSettings(const FString& DevAuthToken)
+{
+	USpatialGDKEditorSettings* GDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>();
+	GDKEditorSettings->SetDevelopmentAuthenticationToken(DevAuthToken);
+	if (GDKEditorSettings->SpatialOSNetFlowType == ESpatialOSNetFlow::CloudDeployment)
```

Comment:
I wonder if we also want the following logic also inside GDKEditorSettings->SetDevelopmentAuthenticationToken

if (GDKEditorSettings->SpatialOSNetFlowType == ESpatialOSNetFlow::CloudDeployment)
{
GDKEditorSettings->SetUseDevelopmentAuthenticationFlow(true);
}

Reply:
No, this is specifically done here because the user may have switched the connection flow to something else while the dev auth token was being generated.